woocommerce-free-shipping-progress-bar-block.php

```diff
@@ -44,13 +44,29 @@ function enqueue_frontend_script() {
 function add_attributes_to_block( $attributes = [], $content = '' ) {
 	$escaped_data_attributes = [];
 
+	// Add the custom background color to the attributes array.
+	if ( ! $attributes['backgroundColor'] && $attributes['style']['color']['background'] ) {
+		$attributes['backgroundColor'] = $attributes['style']['color']['background'];
+	}
+
+	// Add the custom text color to the attributes array.
+	if ( ! $attributes['textColor'] && $attributes['style']['color']['text'] ) {
+		$attributes['textColor'] = $attributes['style']['color']['text'];
+	}
+
```

Comment:
I'd like to understand what's happening here really? this logic seems redundant.

Reply:
Thanks for asking. In https://github.com/woocommerce/woocommerce-free-shipping-progress-bar-block/pull/14#issuecomment-965168997, I added a bit more context on that, but in a nutshell, when saving custom colors, the div on the frontend looks like this:

```
<div data-style="{"color":{"background":"#bf616a","text":"#5e81ac"}}" data-current-total="15" data-free-shipping-from="50" class="wp-block-nielslange-free-shipping-progress-bar">....</div>
```

The code, you pointed out, converts this to:

```
<div data-background-color="#bf616a" data-text-color="#5e81ac" data-current-total="15" data-free-shipping-from="50" class="wp-block-nielslange-free-shipping-progress-bar">...</div>
```

That way, the custom colors can be picked up correctly when rendering this component. It definitely looks redundant. It is a slightly simplified version of https://github.com/WordPress/gutenberg/blob/2d2e51ee954bf26f3e1a1a4187d0913a31f9459d/packages/block-library/src/search/index.php#L311-L352 though, and I cannot think of a more efficient way of adding custom colors as data-attributes. 

Do you have a suggestion on how to simplify this part?

Reply:
> Thanks for asking. In #14 (comment), I added a bit more context on that, but in a nutshell,

Ah yeah, I missed that, thanks for pointing out.

> That way, the custom colors can be picked up correctly when rendering this component. It definitely looks redundant. It is a slightly simplified version of

Handling colors is still a bit tricky right now in Gutenberg, I managed to get this to work another way, using `useBorderProps` but it's experimental. I think this solution is by itself good and does the job, we can probably see where things are in GB later down the road.

Reply:
>  I think this solution is by itself good and does the job, we can probably see where things are in GB later down the road.

That sounds like a good plan.